File: 4.py
max = 0
for c1 in range (100,1000):
    for c2 in range (100,1000):
        s = c1 * c2
        temp=s
        rev=0
        while s>0:
            dig=s%10
            rev=rev*10+dig
            s=s//10 
        if temp==rev and temp > max:
            max = temp
print (max)

File: JavaScript/4.js
var max = 0;
for (i=100;i<1000;i++){
    for (j=100;j<1000;j++){
        var s;
        s = i * j
        temp = s
        rev = 0
        while (s >0){
            var dig = s%10
            rev = rev*10+dig
            s = Math.floor(s/10)
            if (temp === rev & temp > max){
                max = temp
            }
        }
    }
}
console.log(max)